### PR TITLE
Add .357 SIG ammunition and a P226

### PR DESCRIPTION
**Pull request by ymber**
Adds a SIG Sauer P226 chambered for .357 SIG, two varieties of .357 SIG ammunition and reloads, and appropriate itemgroups for all new items.

The massive diff for ammo_types.json is because I ran it through the web linter and the whole file got a load of formatting changes. There's only 6 new lines in there defining a 357sig ammo type.

### CODE REVIEW

File: data/json/items/magazine/357.json [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+		"description": "A 15 round double stack box magazine for a SIG P226 chambered for .357 SIG rounds.",
+		"weight": 150,
+		"volume": 1,
+		"price": 3000,
+		"material": "steel",
+		"symbol": "#",
+		"color": "light_gray",
+		"ammo_type": "357sig",
+		"capacity": 15,
+		"reliability": 5,
```
Comment (BorkBorkGoesTheCode) [inline data/json/items/magazine/357.json]: What's your source for this? Anything less than 6 ingame sucks as much as improvised firearms.

File: data/json/items/generic/casing.json [hunk 2]
```diff
@@ -246,3 +255,5 @@
     "damage_states": [ 0, 0 ]
   }
 ]
+
+
```
Comment (BorkBorkGoesTheCode) [inline data/json/items/generic/casing.json]: What happened here?

File: data/json/items/gun/357.json [hunk 3]
```diff
@@ -14,7 +14,7 @@
     "symbol": "(",
     "color": "dark_gray",
     "ammo": "357sig",
-    "dispersion": 400,
+    "dispersion": 480,
     "durability": 7,
```
Comment (BorkBorkGoesTheCode) [inline data/json/items/gun/357.json]: That's not recoil, that's dispersion. Dispersion affects all shots.

**Comment by kevingranade** [comment]
Please lint separately, no need to generate huge diffs.

File: data/json/items/magazine/357.json (hunk 1 above)
Comment (ymber) [reply data/json/items/magazine/357.json]: Not sure how that got in there. It was supposed to be 9.

File: data/json/items/generic/casing.json (hunk 2 above)
Comment (ymber) [reply data/json/items/generic/casing.json]: I kept getting trailing whitespace like that copying the web linter's output. Forgot to remove it in that file.

File: data/json/items/gun/357.json (hunk 3 above)
Comment (ymber) [reply data/json/items/gun/357.json]: Mistake in the commit message. I meant dispersion.

**Comment by ymber** [comment]
Fixed errata in definitions and json formatting. I'll lint casing.json in a different PR.

File: data/json/items/ammo/357.json [hunk 4]
```diff
@@ -0,0 +1,50 @@
+[
+  {
+    "id": "357sig_fmj",
+    "type": "AMMO",
+    "name": ".357 SIG FMJ",
+    "description": "Jacketed .357 SIG ammunition. The .357 SIG round is a high velocity pistol cartridge, giving it a flatter trajectory than many handgun rounds.",
```
Comment (illi-kun) [inline data/json/items/ammo/357.json]: > . The

Double whitespace required.

File: data/json/items/ammo/357.json [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+    "dispersion": 30,
+    "recoil": 600,
+    "effects": [ "COOKOFF" ]
+  },
+  {
+    "id": "357sig_jhp",
+    "copy-from": "357sig_fmj",
+    "type": "AMMO",
+    "name": ".357 SIG JHP",
+    "description": "Jacketed hollow point .357 SIG ammunition. The .357 SIG round is a high velocity pistol cartridge, giving it a flatter trajectory than many handgun rounds.",
```
Comment (illi-kun) [inline data/json/items/ammo/357.json]: > . The

Double whitespace required.

File: data/json/items/gun/357.json [hunk 6]
```diff
@@ -0,0 +1,22 @@
+[
+  {
+    "id": "p226_357sig",
+    "copy-from": "pistol_base",
+    "type": "GUN",
+    "name": "SIG P226",
+    "description": "A SIG Sauer P226 chambered for .357 SIG. The P226 is a DA/SA, short-recoil operated semi-automatic pistol.",
```
Comment (illi-kun) [inline data/json/items/gun/357.json]: > . The

Double whitespace required.

File: data/json/items/gun/357.json [hunk 7]
```diff
@@ -0,0 +1,22 @@
...
+  {
+    "id": "p226_357sig",
+    "copy-from": "pistol_base",
+    "type": "GUN",
+    "name": "SIG P226",
+    "description": "A SIG Sauer P226 chambered for .357 SIG.  The P226 is a DA/SA, short-recoil operated semi-automatic pistol.",
+    "weight": 800,
+    "volume": 2,
+    "price": 50000,
+    "to_hit": 1,
```
Comment (illi-kun) [inline data/json/items/gun/357.json]: Most of guns I reviewed for comparison have ```to_hit: -1``` and I think this gun should be the same.

**Comment by ymber** [comment]
I set the gun stats against the M9. It's more durable with slightly more reliable magazines. It should be lighter but the M9's weight stat is lower than it should be. Possibly it should be slightly more accurate.

to_hit is +1 based on GAME_BALANCE.md. +1 from the grip, +0 from length, striking surface, and balance.

**Comment by illi-kun** [comment]
You've missed the fact that starting value to modify is '-2' and not '0', so the proper calculation is
to-hit bonus = -2 +1 +0 +0 +0 = -1


[Game Balance file, section 'To-Hit-Bonuses'](https://github.com/CleverRaven/Cataclysm-DDA/blob/master/doc/GAME_BALANCE.md#to-hit-bonuses)
> To-hit bonuses start at '-2' and are modified as follows for weapons that have the following properties:
...

**Comment by illi-kun** [comment]
These test results should be reviewed carefully:
```
-------------------------------------------------------------------------------
unskilled_shooter_accuracy
  an unskilled shooter with an inaccurate pistol
-------------------------------------------------------------------------------
ranged_balance.cpp:214
...............................................................................
ranged_balance.cpp:172: FAILED:
  CHECK( good_stats[2].avg() < 0.1 )
with expansion:
  0.11964f < 0.1
with messages:
  Normal: [ 180 ]
  Linear: [ 30 30 20 154 ]
  Range: 7
  Max aim speed: 91
  Min aim speed: 0
  good_stats[2].n() := 2984 (0xba8)
  good_stats[2].adj_wald_error() := 0.01961f
-------------------------------------------------------------------------------
competent_shooter_accuracy
  a skilled shooter with an accurate pistol
-------------------------------------------------------------------------------
ranged_balance.cpp:238
...............................................................................
ranged_balance.cpp:193: FAILED:
  CHECK( fast_stats[1].avg() > hit_rate )
with expansion:
  0.3815f > 0.5f
with messages:
  Normal: [ 106 ]
  Linear: [ 15 25 10 720.734 ]
  Range: 3
  Max aim speed: 175.5
  Min aim speed: 16.9432
  shooter.weapon.gun_skill().str() := "pistol"
  shooter.get_skill_level( shooter.weapon.gun_skill() ) := 5
  shooter.get_dex() := 10
  to_milliliter( shooter.weapon.volume() ) := 750 (0x2ee)
  fast_stats[1].n() := 173
  fast_stats[1].adj_wald_error() := 0.11831f
===============================================================================
test cases:      60 |      58 passed | 2 failed
```

**Comment by ymber** [comment]
Test failures shouldn't be caused by anything in this PR. None of the new items are used in ranged_balance.cpp tests.

The best accuracy you can get (p226_357sig + 357sig_fmj) using the new items is achievable with existing items.

**Comment by illi-kun** [comment]
Thanks for the note, it makes sense.

**Comment by Leland** [comment]
Yup, we've got a sporadically failing test. FYI @illi-kun #22973

**Comment by illi-kun** [comment]
@Leland Yeah, I've found this PR after ymber's note. Thanks for info anyway.

**Pull request merged**